### ISSUE TITLE
[Vapor 3] Complex queries results in incorrect SQL

### ISSUE THREAD

**Issue by andrewtheis** [vapor/fluent#466]
Given the following model:
``` swift
final class Event: PostgreSQLModel {
    var description: String?
    var endDate: Date?
    var id: Int?
    var name: String
    var startDate: Date
}
```

And the following query:
``` swift
let eventQuery = try Event.query(on: request)
    .range(0...50)

let now = Date()
    
try eventQuery.group(.or) { orGroup in
    try orGroup.filter(\.startDate >= Date.distantPast)
    try orGroup.filter(\.endDate >= Date.distantPast)
}.filter(\.startDate <= Date.distantFuture).group(.or) { orGroup in
    try orGroup.filter(\.endDate >= now)
    try orGroup.filter(\.endDate == nil)
}

let searchTerm = "xyz"
try eventQuery.group(.or) { (orGroup) in
    try orGroup.filter(\.name, .custom(.sql("ILIKE")), .data("%\(searchTerm)%"))
    try orGroup.filter(\.description, .custom(.sql("ILIKE")), .data("%\(searchTerm)%"))
}
```

Results in this SQL:
``` sql
SELECT * 
FROM "events" 
WHERE (
	("events"."start_date" >= $1 OR "events"."end_date" >= $2)
	AND "events"."start_date" <= $3 
	AND (
		("events"."start_date" >= $4 OR "events"."end_date" >= $5) 
		OR "events"."start_date" <= $6 
		OR "events"."end_date" >= $7 
		OR "events"."end_date" IS NULL
	) 
	AND (
		("events"."start_date" >= $8 OR "events"."end_date" >= $9) 
		OR "events"."start_date" <= $10 
		OR (
			("events"."start_date" >= $11 OR "events"."end_date" >= $12) 
			OR "events"."start_date" <= $13 
			OR "events"."end_date" >= $14
			OR "events"."end_date" IS NULL
		) 
		OR "events"."name" ILIKE $15 
		OR "events"."description" ILIKE $16
	)
) 
LIMIT 50 OFFSET 0 
```

When the expected SQL is:
``` sql
SELECT * 
FROM "events" 
WHERE (
	("events"."start_date" >= $1 OR "events"."end_date" >= $2)
	AND "events"."start_date" <= $3 
	AND ("events"."end_date" >= $4 OR "events"."end_date" IS NULL) 
	AND ("events"."name" ILIKE $5 OR "events"."description" ILIKE $6)
) 
LIMIT 50 OFFSET 0 
```

**Comment by tanner0101** [comment]
Keeping this open as a reminder to add a test case.

**Comment by tanner0101** [comment]
This is fixed in #482
